ecommerce/extensions/payment/processors/braintree.py

```diff
@@ -216,9 +213,93 @@ def create_transaction(self, nonce, basket):
             raise GatewayError
 
         else:
-            # TODO Log the error!
-            # for error in result.errors.deep_errors:
-            #     print("attribute: " + error.attribute)
-            #     print("  code: " + error.code)
-            #     print("  message: " + error.message)
+            self.log_deep_errors(result.errors.deep_errors)
             raise GatewayError
+
+    def get_or_create_customer_payment_method(self, nonce, customer):
+        """
+        Retrieve a braintree customer payment method if it
+        does not exist create one
+
+        Arguments:
+            nonce - Braintree payment nonce
+            customer - Braintree Customer instance
+
+        Returns:
+            It will return braintree PaymentMethod instance or
+            throw GatewayError Exception
+        """
+        customer_id = "{}".format(customer.id)
+        try:
+            customer_payment_method = customer.payment_methods[0].token
+        except IndexError:
+            customer_payment_method = ''
+
+        try:
+            result = braintree.PaymentMethod.find(customer_payment_method)
+            token = result.token
+            logger.info('Found existing PaymentMethod [%s] for customer [%s].',
+                        token, customer_id)
+
+        except braintree.exceptions.not_found_error.NotFoundError:
+            result = braintree.PaymentMethod.create({
+                "customer_id": customer_id,
+                "payment_method_nonce": nonce
+            })
+            if result.is_success:
+                token = result.payment_method.token
+                logger.info('Created new PaymentMethod [%s] for customer [%s].',
+                            token, customer_id)
+            else:
+                self.log_deep_errors(result.errors.deep_errors)
+                return GatewayError
+        return token
+
+    def create_subscription(self, nonce, basket):
+
+        customer = self.get_or_create_customer(basket.owner)
+
+        payment_method_token = self.get_or_create_customer_payment_method(nonce, customer)
+
+        result = braintree.Subscription.create({
+            'payment_method_token': payment_method_token,
+            'plan_id': 'edevate_subscription'
```

Comment:
is it some constant id on the brain tree side? if so might be we should create a constant for it in settings